WIP: Add support for images in icon-grids

pleasantone opened the pull request:
The code was so small, I was worried that we'd have a lot of cut-and-paste duplication.

See https://goattrails.pst.org/events/2024/03/baja-california/#attendees
Sources:
https://github.com/goattrails/hb-goattrails/tree/main/content/events/2024-baja-mexico/index.md
https://github.com/goattrails/hb-goattrails/tree/main/content/events/2024-baja-mexico/iattendees.yaml

Parameters added:
```
image:
  Filename:
  any other images/image parameters...
```

razonyang commented:
Thanks, but it doesn't look well (it's a bit of a waste of space on page), and the main reason about separating into new shortcode is that they are used in different purposes. I'd like to that each shortcode is only responsible for doing one job well.

pleasantone commented:
Can you tell me more about the wasted space?
On Jan 11, 2024 at 8:27 PM -0800, Razon Yang ***@***.***>, wrote:
> Thanks, but it doesn't look well (it's a bit of a waste of space on page), and the main reason about separating into new shortcode is that they are used in different purposes. I'd like to that each shortcode is only responsible for doing one job well.
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you authored the thread.Message ID: ***@***.***>

razonyang commented:
![image](https://github.com/hugomods/bootstrap/assets/17720932/ed96ff1f-9466-4725-90a5-b66607385f88)

> Screenshot captured from local.

pleasantone commented:
Ahh, I see, yours looks quite different than mine, but I was also mis-using description and caption instead of title...

when I changed to title and added me-2 (or the #float-start fragment (see wrapperClass issue in images, it looks much better...

<img width="1015" alt="Screenshot 2024-01-12 at 10 50 05 AM" src="https://github.com/hugomods/bootstrap/assets/5339793/0c630294-0266-4d1b-a18d-da85ad34ccaf">

razonyang commented:
I've thought about this, but I don't want to complex the icon-grid shortcode, since:

1. Hard to maintain: the icon-grid, there are some parameters for icon only, such as the `alignment`, when set as `center`, the icon will be align to center, but it also affects the image, it's hard to maintain different parameters (layouts) for both of icon and images.
2. Confused: the `icon-shortcode` name is confused if we support image.

I've created another shortcode (#105) for #100.

![image](https://github.com/hugomods/bootstrap/assets/17720932/78551e9c-0849-4cb1-ab86-057fddc23b5f)
